src/SFA.DAS.EAS.Web/Orchestrators/EmployerAccountTransactionsOrchestrator.cs

```diff
@@ -314,6 +313,36 @@ public EmployerAccountTransactionsOrchestrator(IMediator mediator, ICurrentDateT
             };
         }
 
+        private static ICollection<TransactionLine> AggregateLevyTransactions(ICollection<TransactionLine> transactions)
+        {
+            if (transactions == null || !transactions.Any())
+                return new List<TransactionLine>();
+
+            if (transactions.Count(t => t.TransactionType == TransactionItemType.Declaration) < 2)
+                return transactions;
+
+
+            var levyTransactions = transactions.Where(t => t.TransactionType == TransactionItemType.Declaration)
+                                               .ToArray();
+
+            var aggregatedTransactions = transactions.Except(levyTransactions).ToList();
+
+            //We haven't populated all the values here as the view doesn't support them yet
+            var levyTransaction = new LevyDeclarationTransactionLine
+            {
+                Description = levyTransactions.First().Description,
+                DateCreated = levyTransactions.Max(t => t.DateCreated),
+                Amount = levyTransactions.Sum(t => t.Amount),
+                TransactionType = TransactionItemType.Declaration,
+                TransactionDate = levyTransactions.Max(t => t.TransactionDate)
+            };
```

Comment:
There are four enumerations on levyTransactions here - each requiring a new instance of an enumerator to be created. Code looks clean but could this be done more efficiently?

Reply:
This was mainly a code amount trade off. I could have created two variables and added logic to a single enumeration but thought at the time it wouldn't be a massive performance hit anyway. I'll add the new logic so we have the best performance.

Reply:
Individually no big deal, but what happens when this is scaled up? .NET mem alloc is good but creating objects increases pressure on the garbage collector increasing the frequency of L0 garbage collections - when the whole process stops - every single .NET thread is suspended for what can be quite a few seconds. 
Trade off - things that make code easier such as LINQ often come with a performance penalty. If it were just one loop here but there are four over the same collection.